xbmc/favourites/FavouritesURL.cpp

```diff
@@ -0,0 +1,210 @@
+/*
+ *  Copyright (C) 2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "FavouritesURL.h"
+
+#include "FileItem.h"
+#include "ServiceBroker.h"
+#include "URL.h"
+#include "addons/AddonManager.h"
+#include "addons/IAddon.h"
+#include "guilib/LocalizeStrings.h"
+#include "input/WindowTranslator.h"
+#include "utils/ExecString.h"
+#include "utils/StringUtils.h"
+#include "utils/SystemInfo.h"
+#include "utils/URIUtils.h"
+#include "utils/log.h"
+
+#include <vector>
+
+namespace
+{
+std::string GetActionString(CFavouritesURL::Action action)
+{
+  switch (action)
+  {
+    case CFavouritesURL::Action::ACTIVATE_WINDOW:
+      return "activatewindow";
+    case CFavouritesURL::Action::PLAY_MEDIA:
+      return "playmedia";
+    case CFavouritesURL::Action::SHOW_PICTURE:
+      return "showpicture";
+    case CFavouritesURL::Action::RUN_SCRIPT:
+      return "runscript";
+    case CFavouritesURL::Action::RUN_ADDON:
+      return "runaddon";
+    case CFavouritesURL::Action::START_ANDROID_ACTIVITY:
```

Comment:
shouldn't this be ifdef'd for android?

Reply:
It was not ifdef'd in the origin code  so I thought that has a reason. I guess even though this code path here would be called due to some handcrafted favourites.xml on a Non-Android system for example, it would do simply nothing at the end. So, this ifdef would only clutter the code with no additional benefit.